modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -63,73 +120,211 @@ def _unstrip_protocol(protocol, path):
 
-        Parameters
-        ----------
-        path : str, path object or file-like object
-            Path to dataset.
-        storage_options : dict
-            Parameters for specific storage engine.
+@_inherit_docstrings(Dataset)
+class PyArrowDataset(Dataset):
+    def __init__(self, path, storage_options):
+        self.path = path
+        self.storage_options = storage_options
+        self.dataset = self._init_dataset()
+        self._row_groups_per_file = None
+        self.filesystem, self._files = self._get_files(self.dataset.files)
 
-        Returns
-        -------
-        dataset : ParquetDataset
-            Dataset object of given Parquet file-path.
-        """
+    def _init_dataset(self):  # noqa: GL08
         from pyarrow.parquet import ParquetDataset
 
-        fs, fs_path = cls._get_fs_and_fs_path(path, storage_options)
+        fs, fs_path = self._get_fs_and_fs_path()
         dataset = ParquetDataset(fs_path, filesystem=fs, use_legacy_dataset=False)
         return dataset
 
+    @property
+    def pandas_metadata(self):
+        return self.dataset.schema.pandas_metadata
+
+    @property
+    def columns(self):
+        return self.dataset.schema.names
+
+    @property
+    def engine(self):
+        return "pyarrow"
+
+    def row_groups_per_file(self):
+        from pyarrow.parquet import ParquetFile
+
+        if self._row_groups_per_file is None:
+            row_groups_per_file = []
+            # Count up the total number of row groups across all files and
+            # keep track of row groups per file to use later.
+            for file in self.files:
+                with self.filesystem.open(file) as f:
+                    row_groups = ParquetFile(f).num_row_groups
+                    row_groups_per_file.append(row_groups)
+            self._row_groups_per_file = row_groups_per_file
+        return self._row_groups_per_file
+
+    @property
+    def files(self):
+        if self._files is None:
+            self._files = self._get_files(self.dataset.files)
+        return self._files
+
+    def read_table_as_pandas(
+        self,
+        columns,
+    ):
+        from pyarrow.parquet import read_table
+
+        return read_table(
+            self.path, columns=columns, filesystem=self.filesystem
+        ).to_pandas()
+
+
+@_inherit_docstrings(Dataset)
+class FastParquetDataset(Dataset):
+    def __init__(self, path, storage_options):
+        self.path = path
+        self.storage_options = storage_options
+        self.dataset = self._init_dataset()
+        self._row_groups_per_file = None
+        self.filesystem, self._files = self._get_files(self._get_fastparquet_files())
+
+    def _init_dataset(self):  # noqa: GL08
+        from fastparquet import ParquetFile
+
+        fs, fs_path = self._get_fs_and_fs_path()
+        dataset = ParquetFile(fs_path, fs=fs)
+        return dataset
+
+    @property
+    def pandas_metadata(self):
+        import json
```

Comment:
Isn't this part of the python standard library? In that case we can move it to the top of the file.